### DISCUSSION TITLE
linking issues on Rpi4

### DISCUSSION THREAD

**Discussion by japeart** [ocornut/imgui#5807] [New Users Build/Link/Run issues ONLY!]
I downloaded the libimgui-dev package via apt, and have linked into my gcc makefile using:

-L/usr/lib/arm-linux-gnueabihf -limgui

However, it is complaining about undefined references despite the fact I can see the functions when I run nm on the library file. Any suggestions?

**Comment by ocornut** [comment]
> Any suggestions?

Maybe start by listing the undefined references :)

**Reply by japeart** [reply]
Sure :)

/usr/bin/ld: /usr/lib/arm-linux-gnueabihf/libimgui.a(imgui_draw.o): in function `ImFontAtlasBuildPackCustomRects(ImFontAtlas*, void*)':
(.text+0x7d6c): undefined reference to `stbrp_pack_rects'
/usr/bin/ld: /usr/lib/arm-linux-gnueabihf/libimgui.a(imgui_draw.o): in function `ImFontAtlasBuildWithStbTruetype(ImFontAtlas*)':
(.text+0xac9c): undefined reference to `stbtt_GetFontOffsetForIndex'
/usr/bin/ld: (.text+0xacb0): undefined reference to `stbtt_InitFont'
/usr/bin/ld: (.text+0xaee4): undefined reference to `stbtt_FindGlyphIndex'
/usr/bin/ld: (.text+0xb284): undefined reference to `stbtt_ScaleForPixelHeight'
/usr/bin/ld: (.text+0xb2b4): undefined reference to `stbtt_FindGlyphIndex'
/usr/bin/ld: (.text+0xb2fc): undefined reference to `stbtt_GetGlyphBitmapBoxSubpixel'
/usr/bin/ld: (.text+0xb3e4): undefined reference to `stbtt_PackBegin'
/usr/bin/ld: (.text+0xb434): undefined reference to `stbrp_pack_rects'
/usr/bin/ld: (.text+0xb590): undefined reference to `stbtt_ScaleForMappingEmToPixels'
/usr/bin/ld: (.text+0xb6c0): undefined reference to `stbtt_PackFontRangesRenderIntoRects'
/usr/bin/ld: (.text+0xb6f8): undefined reference to `stbtt_PackEnd'
/usr/bin/ld: (.text+0xb780): undefined reference to `stbtt_ScaleForPixelHeight'
/usr/bin/ld: (.text+0xb798): undefined reference to `stbtt_GetFontVMetrics'
/usr/bin/ld: (.text+0xb87c): undefined reference to `stbtt_GetPackedQuad'
collect2: error: ld returned 1 exit status

**Reply by PathogenDavid** [reply]
The Debian distribution of Dear ImGui is configured to use the version of stb provided by `libstb-dev`.

If you use `pkg-config` instead of manually passing `-limgui` it should automatically pull it in. You can also probably just add `-lstb`.

It's also worth pointing out that the release of Dear ImGui provided by Debian is quite old, v1.81 came out in February 2021. You're probably better off consuming it the normal way.
